src/internal/operators/takeWhile.ts

```diff
@@ -1,8 +1,15 @@
 /** @prettier */
-import { OperatorFunction, MonoTypeOperatorFunction } from '../types';
+import { OperatorFunction, MonoTypeOperatorFunction, Falsy } from '../types';
 import { operate } from '../util/lift';
 import { OperatorSubscriber } from './OperatorSubscriber';
 
+export function takeWhile<T>(predicate: (value: T, index: number) => false, inclusive?: boolean): OperatorFunction<T, never>;
```

Comment:
issue: I don't think this can have a return type of `Operator<T, never>` unless `inclusive` is `false` or not specified:

```ts
export function takeWhile<T>(predicate: (value: T, index: number) => false, inclusive?: false): OperatorFunction<T, never>;
```

Even though the predicate always returns `false`, the operator is going to emit the value the caused the predicate to do so if `inclusive` is `true`.

Reply:
That's correct. Good catch. Updated and added a test.